chromium_src/components/site_engagement/content/site_engagement_service.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_SITE_ENGAGEMENT_CONTENT_SITE_ENGAGEMENT_SERVICE_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_SITE_ENGAGEMENT_CONTENT_SITE_ENGAGEMENT_SERVICE_H_
+
+#define HandleMediaPlaying \
+  UnusedFunction() {}      \
+                           \
+ protected:                \
+  virtual void HandleMediaPlaying
+
+#define HandleNavigation \
+  UnusedFunction2() {}   \
+  virtual void HandleNavigation
+
+#define HandleUserInput \
+  UnusedFunction3() {}  \
+  virtual void HandleUserInput
+
+#define OnEngagementEvent \
+  UnusedFunction4() {}    \
+                          \
+ private:                 \
+  void OnEngagementEvent
```

Comment:
this was a little confusing at first, wouldn't it be simpler to leave them all private and add a friend?

Reply:
addressed in https://github.com/brave/brave-core/commit/4888b9483dba1bfe3d422f0f0bd7a322cac02497